redash/handlers/embed.py

```diff
@@ -55,8 +56,10 @@ def run_query_sync(data_source, parameter_values, query_text, max_age=0):
             query_result, updated_query_ids = models.QueryResult.store_result(data_source.org_id, data_source.id,
                                                                                   query_hash, query_text, data,
                                                                                   run_time, utils.utcnow())
+        else:
+            query_result = models.QueryResult(data=data, data_source=data_source, query_hash=query_hash)
```

Comment:
@arikfr and @whummer ,

There is still an issue here. When `query_result` is None (max_age <= 0), I'm creating this new QueryResult instance but since there's no `id` the csv/xlsx download won't work.